src/riak_kv_qry_worker.erl

```diff
@@ -262,7 +262,31 @@ throttling_spawn_index_fsms(State) ->
 estimate_query_size(#state{n_subqueries_done = NSubqueriesDone} = State)
   when NSubqueriesDone < 2  ->
     State;
-
```

Comment:
We just need to skip the estimate check if this is the last chunk. It is already in memory so anything bad that could happen will have happened already.